lib/knexFunctions.js

```diff
@@ -14,19 +15,133 @@ function createGame(id){
   })
   .returning("id")
   .then(function(gameid){
-    allGames[gameid] = (gops.createGameObject("gops", id));
+    return knex('user_games')
+    .insert({
+      user_id: userid,
+      game_id: gameid[0]
+    }).returning("game_id")
+  }).then(function(gid){
```

Comment:
Can we rename gameid and gid to newGameID in each instance? or just gameid is fine. but the fact that it changes each time makes me think that you are getting something new each time when you're actually just passing it along from function to function.